tool/tsh/db.go

```diff
@@ -124,33 +124,42 @@ func databaseLogin(cf *CLIConf, tc *client.TeleportClient, db tlsca.RouteToDatab
 		// ref: https://redis.io/commands/auth
 		db.Username = defaults.DefaultRedisUsername
 	}
-	profile, err := client.StatusCurrent(cf.HomePath, cf.Proxy)
+
+	// Note: we don't use StatusCurrentWithIdentity here as virtual identity
+	// profiles are read-only.
+	profile, err := client.StatusCurrentWithIdentity(cf.HomePath, cf.Proxy, cf.IdentityFileIn)
```

Comment:
Would it make sense to separate the concept of read-only profile from identity-based profile? Right now both are expressed with catch-all "virtual" designation, but it feels like a leaky abstraction.

Reply:
It might, though I'm not sure what that use-case would be. As far as I know the only place read-only profiles make sense is identity-based profiles: there's no other way to get a certificate that has database/app/k8s/etc fields.

Either you go the usual way through tsh (which is inherently read/write), or you use `tctl auth sign ...` / Machine ID in which case you bring an identity file with whichever additional fields you'd like. A read-only tsh profile wouldn't be able to request e.g. database certs in a very useful way.